mathjax3-ts/input/tex/color/ColorUtil.ts

```diff
@@ -0,0 +1,194 @@
+/*************************************************************
+ *
+ *  Copyright (c) 2018 Omar Al-Ithawi and The MathJax Consortium
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/**
+ * @fileoverview Utility functions and classes for the color package.
+ *
+ * @author i@omardo.com (Omar Al-Ithawi)
+ */
+
+
+import TexError from '../TexError.js';
+import { COLORS } from './ColorConstants.js';
+
+export type ColorModelProcessor = (def: string) => string;
+export type MethodsMap = Record<string, ColorModelProcessor>;
```

Comment:
Should this be a `Map` rather than a `Record` (like `COLORS` is?).

Reply:
@dpvc not really. All of `BaseMethods`, `AmsMethods` and `BoldsymbolMethods` are `Record`s. 

Unless we'd like to refactor all of the others, I'd like to make Color.js consistent with them.

Reply:
The thing that corresponds to `BaseMethods`, etc, is `ColorMethods` in `ColoreMethods.ts` above.  The `ColorMethodProcessors` below is specific to your code, and can be handled however you want.  Historically, we've had to use generic objects (`Record`s) for this type of task because Javascript didn't have anything better, and when user input is used as the property key, it could lead to unexpected results, because general objects have lots of pre-defined keys (like `toString` for example).  This leads to the need to use `hasOwnProperty()`, which is awkward.  You do use it here (line 50), so are safe, but now that Javascript has Map objects for just this type of situation, it seems that one should really use them.  I admit that I don't always remember to do that myself, but I'm trying to move over to maps wherever possible.